js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+                '5003': 'MakeOrderIDFailed',
+                '5004': 'DuplicateOrder',
+                '5005': 'SellLessThanBuy',
+                '5006': 'BuyGreaterThanSell',
+                '5007': 'NotFindOrder',
+                '5008': 'OrderCanceled',
+                '5009': 'WriteTradeFailed',
+                '5010': 'WrongPrice',
+                '5011': 'WrongAmount',
+                '5012': 'WrongUserID',
+                '5013': 'OrderDone',
+                '5014': 'OrderFailed',
+                '5015': 'OrderNonTradable',
+                '5016': 'AmountLessThanLimit',
+                '5017': 'NewOrderDisabled',
+                '5018': 'CancelOrderDisabled',
+                '5019': 'TriggerPriceEqualClosePrice',
+                '5020': 'OrderCountGreaterThanLimit',
+                '9001': 'Key Already Exists',
+                '9002': 'Key Not Exists',
+                '9003': 'Invalid key',
+                '9004': 'Signature Error',
+                '9005': 'Invalid Token',
+                '9006': 'Token Expires',
+                '9007': 'Invalid User Credential',
+                '9008': 'Frequent Operations',
+                '9009': 'IP Disabled',
+                '9010': 'Key Creation Fails',
+                '9011': 'Unauthorized',
+                '9014': 'Frequent Operations',
+                '9015': 'Not in Binded IP',
+                '9016': 'Exceed Maximum Times of Request Allowed in A Single Day',
+                '9017': 'Fail to Obtain New Token',
+                '9018': 'Key Expires',
+                '9019': 'Date Field not Found in Headers',
+                '9020': 'Improper Date Field Found in Headers',
+                '9021': 'Incoming Time not within 15 Minutes',
+                '9022': 'Token Kicked Off',
+            },
+            'options': {
+                'change_quote': {
+                    'USDT': 'USD',
+                },
+                'order_type': {
+                    '0': '_',
+                    '1': 'Buy',
+                    '2': 'Sell',
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const result = [];
+        const spotResponse = await this.publicGetSymbolAll2 ();
+        const spotMarkets = this.safeValue (spotResponse, 'data', {});
+        const markets = this.safeValue (spotMarkets, 'list', []);
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = market[1];
+            const parts = id.split ('_');
+            const baseId = parts[0];
+            const quoteId = parts[1];
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const precision = {
+                'amount': market[7],
+                'price': market[5],
+            };
+            const symbol = base + '/' + quote;
+            const taker = this.fees['trading']['taker'];
```

Comment:
We don't need to set this manually, because `this.fees['trading']` extends each market by default.